### PR TITLE
VFB-114: Table row styling

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Add highlight on hover
- Add pointer on hover for parcels, clients, client detail modal tables
- Parcels table checkbox cell on click does not trigger opening of modal

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA 
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
  - Will also be automatically checked on pull request
- [x] Make sure you've tested via `npm run test`
  - Will also be automatically checked on pull request

### CODE REVIEW

File: src/app/parcels/ParcelsPage.tsx [hunk 1]
```diff
@@ -322,9 +322,14 @@ const ParcelsPage: React.FC<{}> = () => {
         lastStatus: rowToLastStatusColumn,
     };
 
-    const onParcelTableRowClick = (row: Row<ParcelsTableRow>): void => {
-        setSelectedParcelId(row.data.parcelId);
-        router.push(`/parcels?${parcelIdParam}=${row.data.parcelId}`);
+    const onParcelTableRowClick = (row: Row<ParcelsTableRow>, event: React.MouseEvent): void => {
+        const target = event.target as HTMLDivElement;
```
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:326]: I think it's quite hard to ensure this is indeed HTMLDivElement (in fact i'm not sure now). Let's keep it general for now
```suggestion
        const target = event.target as HTMLElement;
```

File: src/app/parcels/ParcelsPage.tsx [hunk 2]
```diff
@@ -322,9 +322,14 @@ const ParcelsPage: React.FC<{}> = () => {
...
+        if (target.children.length > 0 && target.children[0].tagName === "INPUT") {
```
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:327]: What are we really checking here?

File: src/components/Tables/Table.tsx [hunk 3]
```diff
@@ -368,17 +371,20 @@ const StyledIcon = styled(Icon)`
 
 const TableStyling = styled.div`
     // the component with the filter bars
+
```
Comment (rickaw) [inline src/components/Tables/Table.tsx:374]: nitpick: Are these empty lines intentional?

**Review by rickaw — COMMENTED** [review]
Great! Though I don't think the images on the PR are showing any diff? Am i being blind?

**Comment by minalui** [comment]
> Great! Though I don't think the images on the PR are showing any diff? Am i being blind?

The first row is highlighted! Printscreen doesn't show the mouse though unfortunately.

File: src/app/parcels/ParcelsPage.tsx (hunk 2 above)
Comment (minalui) [reply src/app/parcels/ParcelsPage.tsx:327]: We are checking that the table cell that was clicked has children and the first child is an input. The row has two cells that have children elements (cell 1 and cell 5), the others do not have children.

image
* I really want to check if data-column-id is equal to 1 but not sure how to achieve this in an elegant way. Any suggestions?

Comment (minalui) [reply src/app/parcels/ParcelsPage.tsx:327]: Maybe checking id begins with "cell-1"?

File: src/app/parcels/ParcelsPage.tsx [hunk 4]
```diff
@@ -322,9 +322,14 @@ const ParcelsPage: React.FC<{}> = () => {
         lastStatus: rowToLastStatusColumn,
     };
 
-    const onParcelTableRowClick = (row: Row<ParcelsTableRow>): void => {
-        setSelectedParcelId(row.data.parcelId);
-        router.push(`/parcels?${parcelIdParam}=${row.data.parcelId}`);
+    const onParcelTableRowClick = (row: Row<ParcelsTableRow>, event: React.MouseEvent): void => {
+        const target = event.target as HTMLElement;
+        if (target.id.startsWith("cell-1")) {
```
Comment (rickaw) [inline src/app/parcels/ParcelsPage.tsx:327]: I don't think this is specific enough to target the checkbox. It wouldn't be easy to spot this if we decide to show an icon as the first column for example.

Seems like there's a `ignoreRowClick` flag in the DataTable column. Could we use this, and define the checkbox column to ignore row clicks?
https://react-data-table-component.netlify.app/?path=/docs/api-columns--docs

File: src/components/Tables/Table.tsx [hunk 5]
```diff
@@ -379,6 +382,7 @@ const TableStyling = styled.div`
         background-color: transparent;
 
         // the pagination bar
+
```
Comment (rickaw) [inline src/components/Tables/Table.tsx:385]: there's an extra empty line here too :)

File: src/app/parcels/ParcelsPage.tsx (hunk 4 above)
Comment (minalui) [reply src/app/parcels/ParcelsPage.tsx:327]: Think I need some help on this

Comment (minalui) [reply src/app/parcels/ParcelsPage.tsx]: I realised I can set ignoreRowClick on checkboxconfig

**Comment by minalui** [comment]
I found that there is one column called iconsColumn (flagged for attention/requires followup call) that is not being displayed - just wondered if there was some other uses for it?

**Review by rickaw — APPROVED** [review]
Nice one thank you!

**Pull request merged**